models/ed/R/sample.IC.ED2.R

```diff
@@ -0,0 +1,114 @@
+##' Samples intial conditions for ED2
+##' 
+##' NOTE: almost all of the fcn args can be extracted from pecan settings list, just wnated to be explicit
+##' 
+##' @param samples a dataframe with columns: tree id, spp info, Measurement_Year, DBH, plot id (if applicable), could come from load_data
+##' @param format_name how the spp info provided: usda, fia, latin_name, custom
+##' @param start_date settings$run$start.date
+##' @param pfts settings$pfts
+##' @param ne number of ensembles requested
+##' @param latitude settings$run$site$lat
+##' @param longitude settings$run$site$lon
+##' @param source the source of the samples, will appear as prefix in the IC filenames e.g. "HF_lyford.PalEON"
+##' @param metadata any known metadata: age, area (to calculate density) etc.
+##' @param outfolder settings$rundir
+##' @param host_info host info, e.g. settings$host
+##' @param inputs_path dir path to inputs, e.g. dirname(settings$run$inputs$css$path)
+##' 
+##' @return IC 
+##' 
+##' @author Istem Fer
+##' @export
+##' 
+sample.IC.ED2 <- function(samples, format_name = "usda", start_date, pfts, ne, 
+                          latitude, longitude, source, metadata, outfolder, host_info, inputs_path) {
+  
+
+  
+  year <- lubridate::year(start_date)-1 # year before the initial run
+  
+  # subset samples for the year, if coming from load_data won't need this
+  samples <- samples[samples$Measurement_Year == year, ]
+  
+  # remove rows with NAs (we don't want DBH to be NA but do we want to allow missing taxa?)
+  samples <- samples[complete.cases(samples), ]
+
+  # match spp info with species names 
+  # the matching can go into the ic-loop below, depending which is easier to handle sample size vs ensemble size
+  # for sample sizes of 10^5 this is still fast without sampling
+  spp_match <- match_species_id(input_codes = samples[[format_name]], format_name = format_name)
+  
+  # match pfts
+  pft_match   <- match_pft(bety_species_id = spp_match$bety_species_id, pfts = pfts) 
```

Comment:
At a high level, this function seems to replicate, rather than build on, the existing veg workflow. This first bit is redundant. The middle bit that does the sampling isn't actually model specific, but should apply to all cohort models (and generalizable to all pool-based models using the cohort2pool function). The last bit seems easily generalizable to other veg2models.

Reply:
my only concern with complete integration with IC workflow is that IC workflow currently works for generating one IC file per site from one data source (naming files and inserting to the DB accordingly), but in this case we are generating multiple files for a single site from a single dataframe. I'll check how I can do that

on other ensemble naming and DB insertion, what have you decided? currently I'm not writing anything to the DB, I was just writing files like `HF_lyford.PalEON_ens*.1961.lat42.53lon-72.18.css` to a folder

Reply:
Yep, understood. @Luke-Dramko just tacked that problem for met.process and generalized `convert.input` to work with ensembles. 

The relevant code starts here:
https://github.com/PecanProject/pecan/blob/1041156104b544d0d31024f0028d75c66aac7eea/base/utils/R/convert.input.R#L34

An example of that being called is here:
https://github.com/PecanProject/pecan/blob/1041156104b544d0d31024f0028d75c66aac7eea/modules/data.atmosphere/R/download.raw.met.module.R

And here for an example of the downstream workflow using the ensemble (where the default of n=1 is just a special case):
https://github.com/PecanProject/pecan/blob/develop/modules/data.atmosphere/R/met.process.R#L253